### DISCUSSION TITLE
How to approach clickable checklist items?

### DISCUSSION THREAD

**Discussion by johannesjo** [markedjs/marked#3494] [Q&A]
I would like to make my checklist `- [x] bla bla` clickable so that the click toggles the checkmark. I tried adding a `onclick` handler to the `listitem` renderer, but that is stripped away. Any suggestions how I can approach this?

**Comment by UziTech** [comment]
You can override the `checkbox` renderer and remove the disabled attribute

https://github.com/markedjs/marked/blob/17ca80ca2756ce083e43fa30f3f110cbbcf9bdaa/src/Renderer.ts#L101-L105

**Reply by johannesjo** [reply]
Thanks for your reply!! The problem is that I also need to change the underlying model. Changing the view value would not be enough unfortunately.

**Reply by UziTech** [reply]
If you need to add a click handler you can do that after running through marked. Something like:

```js
document.body.append(marked.parse(markdown));

for(const checkbox of document.queryAll("li input[type=checkbox]")) {
  checkbox.addEventListener("change", doSomething);
}
```

**Reply by johannesjo** [reply]
Thank you. It is a bit more complicated in my case as I am using angular and ngx-markdown, but that is probably the approach I need to go for.

**Reply by johannesjo** [reply]
I build the logic for attaching a click handler. Does anybody have any suggestions on how to identify and change the relevant markdown?

**Reply by johannesjo** [reply]
This seems to work:
```TypeScript
private _handleCheckboxClick(ev: MouseEvent): void {
    const allCheckboxes =
      this.previewEl?.element.nativeElement.querySelectorAll('.checkbox-wrapper');

    const checkIndex = Array.from(allCheckboxes || []).findIndex(
      (el) => el === ev.target,
    );
    if (checkIndex !== -1 && this._model) {
      // Find all to-do items in the markdown string
      const todoItems = this._model.match(/- \[[ x]\] .*/g);

      if (todoItems && todoItems[checkIndex]) {
        // Replace the value at checkIndex with the opposite value
        const updatedItem = todoItems[checkIndex].includes('[ ]')
          ? todoItems[checkIndex].replace('[ ]', '[x]')
          : todoItems[checkIndex].replace('[x]', '[ ]');

        // Update the markdown string
        this._model = this._model.replace(todoItems[checkIndex], updatedItem);
        console.log(this._model);
      }
    }
  }
```
